rayon-futures/src/lib.rs

```diff
@@ -156,6 +157,142 @@ struct ScopeFutureContents<'scope, F, S>
     canceled: bool,
 }
 
+impl<'scope, F, S> fmt::Debug for ScopeFutureContents<'scope, F, S>
+    where F: Future + Send + 'scope, S: ScopeHandle<'scope> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.debug_struct("ScopeFutureContents").finish()
+    }
+}
+
+// Newtype so we can implement Into<UnsafeNotify> even though the contents are not 'static.
+#[derive(Debug)]
+struct ArcScopeFuture<'scope, F, S>(Arc<ScopeFuture<'scope, F, S>>)
+where
+    F: Future + Send + 'scope,
+    S: ScopeHandle<'scope>;
+
+impl<'scope, F, S> Clone for ArcScopeFuture<'scope, F, S>
+where
+    F: Future + Send + 'scope,
+    S: ScopeHandle<'scope>,
+{
+    fn clone(&self) -> Self {
+        ArcScopeFuture(self.0.clone())
+    }
+}
+
+impl<'scope, F, S> Notify for ArcScopeFuture<'scope, F, S>
+where
+    F: Future + Send + 'scope,
+    S: ScopeHandle<'scope>,
+{
+    fn notify(&self, id: usize) {
+        self.0.notify(id)
+    }
+
+    fn clone_id(&self, id: usize) -> usize {
```

Comment:
I don't think you really need to implement clone_id and drop_id, since you don't ever use a meaningful id that needs to be cloned or dropped specially.